src/ring_theory/graded_algebra/homogeneous_ideal.lean

```diff
@@ -471,3 +469,67 @@ lemma ideal.homogeneous_hull_eq_Inf (I : ideal A) :
 eq.symm $ is_glb.Inf_eq $ (ideal.homogeneous_hull.gc 𝒜).is_least_l.is_glb
 
 end galois_connection
+
+section irrelevant_ideal
+
+open graded_algebra set_like.graded_monoid direct_sum
+
+variables [comm_semiring R] [semiring A]
+variables [algebra R A] [decidable_eq ι]
+variables [canonically_ordered_add_monoid ι]
+variables (𝒜 : ι → submodule R A) [graded_algebra 𝒜]
+
+/--
+If `A` is graded by a canonically ordered add monoid, then the projection map `x ↦ x₀` is a ring
+homomorphism.
+-/
+def graded_algebra.proj_zero_ring_hom : A →+* A :=
```

Comment:
```suggestion
@[simps]
def graded_algebra.proj_zero_ring_hom : A →+* A :=
```

Reply:
Also. I think this def maybe belongs in `basic.lean`.

Reply:
I have moved it to `basic.lean`